src/Sylius/Bundle/CoreBundle/Security/UserImpersonator.php

```diff
@@ -40,12 +40,24 @@ public function __construct(SessionInterface $session, string $firewallContextNa
 
     public function impersonate(UserInterface $user): void
     {
-        $token = new UsernamePasswordToken(
-            $user,
-            $user->getPassword(),
-            $this->firewallContextName,
-            array_map(/** @param object|string $role */ static function ($role): string { return (string) $role; }, $user->getRoles())
-        );
+        /**
+         * @deprecated parameter credential was deprecated in Symfony 5.4, so in Sylius 1.11 too, in Sylius 2.0 providing 4 arguments will be prohibited.
+         */
+        if (3 === (new \ReflectionClass(UsernamePasswordToken::class))->getConstructor()->getNumberOfParameters()) {
+            $token = new UsernamePasswordToken(
+                $user,
+                $this->firewallContextName,
+                array_map(/** @param object|string $role */ static function ($role): string { return (string) $role; }, $user->getRoles())
+            );
+        } else {
+            $token = new UsernamePasswordToken(
+                $user,
+                null,
```

Comment:
shouldn't it stay like before?
```suggestion
                $user->getPassword(),
```